### ISSUE TITLE
IOS: Email compose not opening when called from modal

### ISSUE THREAD

**Issue by FBorgonjen** [EddyVerbruggen/nativescript-email#30]
When trying to open the email window from within a modal window we are getting an error and no email window. Our app is using a `TabView` as bootstrap component. From the tabview component a modal is opened showing a login page. From this page we have a button which calls the `email.compose` method. The error we get is:
`Warning: Attempt to present <MFMailComposeViewController: 0x103cfc400> on <UITabBarControllerImpl: 0x103caac00> whose view is not in the window hierarchy!`

It looks like there is a clash betwee the window that opens the login-modal and the email-window. 

If we call the exact same code from outside a modal window (inside one of the tabview pages) the code works as expected.

We are on:
- tns: 4.2.2
- nativescript-angular 6.0.6
- tns-core-modules: 4.2.0

**Comment by EddyVerbruggen** [comment]
Interesting find. Try playing around with this line in node_modules/nativescript-email.. perhaps you can find a way to fix it:

https://github.com/EddyVerbruggen/nativescript-email/blob/6d6880c5db755628eb1ef76ef40bc9d2105fa4ec/email.ios.js#L97

**Comment by FBorgonjen** [comment]
Thanks for your quick response. I will fiddle with this piece of code and post my findings here

**Comment by Robsta84** [comment]
It's not the default behavior of an iOS app to call the mail composer from a modal view. A workaround is to dismiss the modal view after tapping on the mail button and in the closeCallback() you can now open the mail composer.

**Comment by msaravitz** [comment]
Fixed in version 1.5.4

**Comment by buuhuu** [comment]
This only works when the modal was opened using the topmost() frame as parent. If it was opened differently it will fail. For example we open a Modal based on an event in our AppComponent

```
const options: ModalDialogOptions = {
  context: { config },
   fullscreen: true,
   viewContainerRef: this._vcRef, // ViewContainerRef injected into AppComponent
   target: topmost() // this needs to be added otherwise it doesn't work 
};

this._modalService.showModal(MyModalComponent, options);
```
